tools/skylark/pybind.bzl

```diff
@@ -375,22 +391,84 @@ generate_pybind_documentation_header = rule(
         "targets": attr.label_list(
             mandatory = True,
         ),
-        "headers_without_attic": attr.label(
-            allow_files = True,
-            mandatory = False,
-        ),
+        "target_deps": attr.label_list(default = []),
         "_mkdoc": attr.label(
             default = Label("//tools/workspace/pybind11:mkdoc"),
             allow_files = True,
             cfg = "host",
             executable = True,
         ),
         "out": attr.output(mandatory = True),
-        "out_xml": attr.output(mandatory = True),
+        "out_xml": attr.output(mandatory = False),
         "root_name": attr.string(default = "pydrake_doc"),
         "exclude_hdr_patterns": attr.string_list(),
     },
     fragments = ["cpp"],
     implementation = _generate_pybind_documentation_header_impl,
     output_to_genfiles = True,
 )
+
+def add_pybind_coverage_data(
+        name = "pybind_coverage_data",
+        subpackages = []):
+    """Gathers necessary source files so that we can have access to them for
+    coverage analysis (Bazel does not like inter-package globs). This should be
+    added to each package where coverage is desired."""
+    native.filegroup(
+        name = name,
+        srcs = native.glob(["*_py*.cc"]),
```

Comment:
Do you think we're following a convention in naming the bindings' files?
Although, `find ./bindings/pydrake \( -type f -o -type l \) -name '*.cc' | grep -v '.*_py.*\.cc'`
gives only a couple of test files, I don't know if we're following a convention because, there are even files like this:

`./bindings/pydrake/systems/framework_py_systems.cc`  in addition to the `.*_py\.cc` files.

Reply:
Yes, that is the conventions that I've tried to encourage / use. For example:
```
$ find . -name '*_py*.cc' | sort
...
./systems/framework_py.cc
./systems/framework_py_semantics.cc
./systems/framework_py_systems.cc
./systems/framework_py_values.cc
...
```
Shows everything that I want?

(and the test files would be ignored, because the glob is explicitly non-recursive)

Reply:
Yeah, that's what I was saying, that although `'.*_py.*\.cc'` takes care of all the files, there's lack of consistent convention.

Reply:
That actually is done explicitly, because they don't declare a module on their own (but instead are used by `framework_py.cc`. And it's more or less captured, but I can just make a PR to capture this part of the convention.

(Also, I'm still not sure why this is a perceived problem? It captures the relevant files, no?)

Reply:
No, I didn't mean it as a problem.  I just wanted to make sure that there is a convention and we don't miss the convention in future files, since we're hard coding it.

I had noticed that I had missed `framework_py_semantics.cc` type files while using globs (because I'd only used `*_py.cc`).  Anyway, I'm fixing the TODOs and will add it to my PR.  Thanks!

Reply:
Sweet! I just wasn't sure what the call to action was. Perhaps "Should this convention be documented" would've sufficed?

Submitted convention doc update here: https://github.com/RobotLocomotion/drake/pull/12403